Composer/packages/client/src/components/GetStarted/GetStartedNextSteps.tsx

```diff
@@ -0,0 +1,310 @@
+  const botProject = useMemo(() => botProjects.find((b) => b.projectId === projectId), [botProjects, projectId]);
+  const [displayManageLuis, setDisplayManageLuis] = useState<boolean>(false);
+  const [displayManageQNA, setDisplayManageQNA] = useState<boolean>(false);
+
+  const { setSettings } = useRecoilValue(dispatcherState);
+  const rootBotProjectId = useRecoilValue(rootBotProjectIdSelector) || '';
+  const settings = useRecoilValue(settingsState(projectId));
+  const mergedSettings = mergePropertiesManagedByRootBot(projectId, rootBotProjectId, settings);
+  const [requiredNextSteps, setRequiredNextSteps] = useState<NextSteps[]>([]);
+  const [recommendedNextSteps, setRecommendedNextSteps] = useState<NextSteps[]>([]);
+  const [optionalSteps, setOptionalSteps] = useState<NextSteps[]>([]);
+
+  const [highlightLUIS, setHighlightLUIS] = useState<boolean>(false);
+  const [highlightQNA, setHighlightQNA] = useState<boolean>(false);
+
+  const hideManageLuis = () => {
+    setDisplayManageLuis(false);
+  };
+  const hideManageQNA = () => {
+    setDisplayManageQNA(false);
+  };
+
+  const openLink = (link) => {
+    TelemetryClient.track('GettingStartedLinkClicked', { method: 'link', url: link });
+    navigateTo(link);
+  };
+
+  const doNextStep = (currentStep) => {
+    const nextAction = requiredNextSteps.filter((f) => !f.checked && f.key != currentStep)?.[0];
+    if (!nextAction?.checked && nextAction?.highlight) {
+      nextAction.highlight();
+    }
+    if (!nextAction) {
+      props.onDismiss();
+      props.onBotReady();
+    }
+  };
+
+  const updateLuisSettings = (newLuisSettings) => {
+    setSettings(projectId, {
+      ...mergedSettings,
+      luis: { ...mergedSettings.luis, ...newLuisSettings },
+    });
+  };
+
+  const updateQNASettings = (newQNASettings) => {
+    setSettings(projectId, {
+      ...mergedSettings,
+      qna: { ...mergedSettings.qna, ...newQNASettings },
+    });
+  };
+
+  const linkToPackageManager = `/bot/${projectId}/plugin/package-manager/package-manager`;
+  const linkToConnections = `/bot/${projectId}/botProjectsSettings/#connections`;
+  const linkToPublishProfile = `/bot/${projectId}/botProjectsSettings/#addNewPublishProfile`;
+  const linkToLUISSettings = `/bot/${projectId}/botProjectsSettings/#luisKey`;
+  const linktoQNASettings = `/bot/${projectId}/botProjectsSettings/#qnaKey`;
+  const linkToLGEditor = `/bot/${projectId}/language-generation`;
+  const linkToLUEditor = `/bot/${projectId}/language-understanding`;
+
+  useEffect(() => {
+    const newNextSteps: NextSteps[] = [];
+    const newRecomendedSteps: NextSteps[] = [];
+
+    const hasLUIS =
+      botProject?.setting?.luis?.authoringKey && botProject?.setting?.luis?.authoringRegion ? true : false;
+    const hasQNA = botProject?.setting?.qna?.subscriptionKey ? true : false;
+
+    const hasPublishingProfile =
+      botProject?.setting?.publishTargets && botProject?.setting?.publishTargets?.length > 0 ? true : false;
```

Comment:
I'm wary of type-unsafe stuff like this, but in this case I think it's okay - I won't block the PR on it since I think there are no cases where those values could be real but falsey.